utils/train_utils.py

```diff
@@ -83,7 +83,9 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
         with MemoryTrace() as memtrace:  # track the memory usage
             model.train()
             total_loss = 0.0
-            for step, batch in enumerate(tqdm(train_dataloader,colour="blue", desc=f"Training Epoch{epoch}")):
+            total_length = len(train_dataloader)//gradient_accumulation_steps
+            pbar = tqdm(colour="blue", desc=f"Training Epoch: {epoch}", total=total_length)
```

Comment:
thanks @mreso shall we do the same for eval loop as well to be consistent, separating the tqdm setting from the progress bar?

Reply:
@HamidShojanazeri The eval loop does not have a print statement in it that breaks the bar. There is also no difference in model and gradient accumulation steps in eval. So I think we can leave it as it as long as we do not add any prints.